src/robot/libraries/Collections.py

```diff
@@ -835,6 +961,29 @@ def _validate_dictionary(self, dictionary, position=1):
             raise TypeError(f"Expected argument {position} to be a dictionary or "
                             f"dictionary-like, got {type_name(dictionary)} instead.")
 
+    def _normalize(self,value):
+        if is_dict_like(value):
+            return {self._normalize(k): self._normalize(value[k]) for k in value}
+        if is_list_like(value):
+            return [self._normalize(v) for v in value]
+        if is_string(value):
+            return value.lower()
+        return value
+
+    def _is_ignore_case(self,ignore_case):
+        if self._normalize(ignore_case)=="key":
+            ignore_case_key=True
+            ignore_case_value=False
+        elif self._normalize(ignore_case)=="value":
+            ignore_case_key=False
+            ignore_case_value=True
+        elif self._normalize(ignore_case)=="both" or ignore_case is True:
+            ignore_case_key=True
+            ignore_case_value=True
+        else:
+            ignore_case_key=False
+            ignore_case_value=False
+        return {'key': ignore_case_key, 'value': ignore_case_value}
```

Comment:
Both of these helper methods can be removed and normalization handled fully by the `Normalizer` class.